odl/solvers/advanced/proximal_operators.py

```diff
@@ -124,14 +133,184 @@ def cconj_prox_factory(step_size):
         Returns
         -------
         proximal : `Operator`
-            The proximal operator of ``prox[a * f^*](x)``
+            The proximal operator of ``s * F^*`` where ``s`` is the step size
         """
-        prox_other = step_size * proximal(1.0 / step_size) * (1.0 / step_size)
+        prox_other = (step_size * prox_factory(1.0 / step_size) *
+                      (1.0 / step_size))
         return IdentityOperator(prox_other.domain) - prox_other
 
     return cconj_prox_factory
 
 
+def proximal_translation(prox_factory, y):
+    """Calculate the proximal of the translated function F(x - y).
+
+    This is calculated according to the rule
+
+        prox[s * F( . - y)](x) = y + prox[s * F](x - y)
+
+    where ``y`` is the translation, and ``s`` is the step size.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``.
+    y : Element in domain of ``F``.
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [CP2011c]_.
+    """
+
+    def translation_prox_factory(step_size):
+        """Create proximal for the translation with a given step_size.
+
+        Parameters
+        ----------
+        step_size : positive `float`
+            Step size parameter
+
+        Returns
+        -------
+        proximal : `Operator`
+            The proximal operator of ``s * F( . - y)`` where ``s`` is the
+            step size
+        """
+
+        return (ConstantOperator(y) + prox_factory(step_size) *
+                (IdentityOperator(y.space) - ConstantOperator(y)))
+
+    return translation_prox_factory
+
+
+def proximal_arg_scaling(prox_factory, scaling):
+    """Calculate the proximal of function F(x * scaling).
+
+    This is calculated according to the rule
+
+        prox[s * F( . * scaling)](x) =
+        1/scaling * prox[s * scaling^2 * F ](x * scaling)
+
+    where ``scaling`` is the scaling parameter, and ``s`` is the step size.
+
+    Parameters
+    ----------
+    prox_factory : `callable`
+        A factory function that, when called with a step size, returns the
+        proximal operator of ``F``
+    scaling : non-zero `float`
+        Scaling parameter
+
+    Returns
+    -------
+    prox : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    For reference on the identity used, see [CP2011c]_.
+    """
+
+    scaling = float(scaling)
+
+    if scaling == 0:
+        return proximal_zero(prox_factory(1.0).domain)
```

Comment:
dislike this way of fishing out the space, but perhaps we'll have to live with it. The option is to move this into the `arg_scaling_prox_factory`.